src/util/string_constant.h

```diff
@@ -11,19 +11,37 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include "std_expr.h"
 
-class string_constantt : public nullary_exprt
+class string_constantt : public constant_exprt
```

Comment:
This is weird: we have a constant_exprt per the C++ type system, but it won’t have ID_constant in the irept type hierarchy. I’m not yet convinced this is the right thing to do.

Reply:
ok, going back to `nullary_exprt`